cpp/CMake/CheckSupportsPreset.cmake

```diff
@@ -0,0 +1 @@
+cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
```

Comment:
Why don't we just update the root `CMakeLists.txt` with this and remove this file?

Reply:
For compat with our internal Docker image?